File: pset6/readability/readability.py
from cs50 import get_string

text = get_string("Text: ")
letters = 0
words = 1
sentences = 0

for i in range(len(text)):
    if text[i].isalpha():
        letters += 1
    if text[i].isspace():
        words += 1
    if text[i] == '.' or text[i] == '?' or text[i] == '!':
        sentences += 1

L = float(letters / words * 100)
S = float(sentences / words * 100)
index = round(0.0588 * L - 0.296 * S - 15.8)

if index >= 16:
    print("Grade 16+")
elif index < 1:
    print("Before Grade 1")
else:
    print(f"Grade {index}")

File: pset2/Readability/readability.c
#include <stdio.h>
#include <cs50.h>
#include <ctype.h>
#include <string.h>
#include <math.h>

//Variables definition
string text;
int letters = 0;
int words = 1;
int sentences = 0;

int main(void)
{
    text = get_string("Text:  "); //User input
    //Count how many letter
    for (int i = 0; i < strlen(text); i++)
    {
        if isalpha(text[i])
        {
            letters ++;
        }
    }
    //Count how many words
    for (int j = 0; j < strlen(text); j++)
    {
        if isspace(text[j])
        {
            words++;
        }
    }
    //Count how many sentences
    for (int k = 0; k < strlen(text); k++)
    {
        if (text[k] == '.' || text[k] == '?' || text[k] == '!')
        {
            sentences++;
        }
    }
    //Do Coleman-Liau index
    float L = (float) letters / words * 100;
    float S = (float) sentences / words * 100;
    float index = round(0.0588 * L - 0.296 * S - 15.8);

    if (index >= 16)
    {
        printf(" Grade 16+\n");
    }
    else if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %i\n", (int) index);
    }
}